serialization_test.go

```diff
@@ -259,36 +267,54 @@ func (errorWriter) Header() http.Header {
 	return http.Header{}
 }
 
-func TestSendYaml(t *testing.T) {
+func TestSendYAML(t *testing.T) {
 	t.Run("success", func(t *testing.T) {
 		w := httptest.NewRecorder()
-		SendYAML(w, response{Message: "Hello World", Code: 200})
+		SendYAML(w, nil, response{Message: "Hello World", Code: http.StatusOK})
 
 		require.Equal(t, "message: Hello World\ncode: 200\n", w.Body.String())
 	})
 
 	t.Run("error", func(t *testing.T) {
 		errorWriter := &errorWriter{}
-		SendYAML(errorWriter, response{Message: "Hello World", Code: 200})
+		SendYAML(errorWriter, nil, response{Message: "Hello World", Code: http.StatusOK})
 		require.Contains(t, errorWriter.arg, "Cannot serialize returned response to YAML")
 	})
 }
 
+func TestSendYAMLError(t *testing.T) {
+	w := httptest.NewRecorder()
+	SendYAMLError(w, nil, errors.New("Hello World"))
+
+	require.Equal(t, http.StatusInternalServerError, w.Result().StatusCode)
+	require.Equal(t, "application/x-yaml", w.Header().Get("Content-Type"))
+	require.Equal(t, "Hello World\n", w.Body.String())
+}
+
 func TestSendJSON(t *testing.T) {
 	t.Run("success", func(t *testing.T) {
 		w := httptest.NewRecorder()
-		SendJSON(w, response{Message: "Hello World", Code: 200})
+		SendJSON(w, nil, response{Message: "Hello World", Code: http.StatusOK})
 
 		require.Equal(t, crlf(`{"message":"Hello World","code":200}`), w.Body.String())
 	})
 
 	t.Run("error", func(t *testing.T) {
 		errorWriter := &errorWriter{}
-		SendJSON(errorWriter, response{Message: "Hello World", Code: 200})
+		SendJSON(errorWriter, nil, response{Message: "Hello World", Code: http.StatusOK})
 		require.Contains(t, errorWriter.arg, "Cannot serialize returned response to JSON")
 	})
 }
 
+func TestSendHTMLError(t *testing.T) {
```

Comment:
I think  TestSendHTML should be added before  TestSendHTMLError, because I don't see it.

maybe I'm wrong, and then it's only a matter of moving this test aside  TestSendHTML

Reply:
I didn't affect the logic with SendHTML. However, I agree, but it should be test with an additional PR/commit. That's a bit more focused. 

Just my opinion. I can add it if you prefer here.